### PR TITLE
Create a basic TypeScript example

### PR DESCRIPTION
**Pull request by mnapoli**
This follows the discussion in https://github.com/serverless/examples/pull/681#issuecomment-1052007939

This is a basic TypeScript example based on the basic Node (JS) example. It uses the `esbuild` plugin as discussed in https://github.com/serverless/serverless/issues/10700

### CODE REVIEW

File: aws-node-typescript/package-lock.json [hunk 1]
```diff
@@ -0,0 +1,3550 @@
+{
```
Comment (pgrzesik) [inline aws-node-typescript/package-lock.json:1]: Do you think we should include the lock file in the example? I think we should not include it

**Review by pgrzesik — COMMENTED** [review]
It's looking good, I just have one minor question and we're good to go 👍

File: aws-node-typescript/package-lock.json (hunk 1 above)
Comment (mnapoli) [reply aws-node-typescript/package-lock.json:1]: Oh no you're right! We'll never be able to keep it up to date.

Maybe we should gitignore all package-lock.json (there are some committed). Anyway it should be fixed in this PR.

**Review by pgrzesik — APPROVED** [review]
👍

**Pull request merged**